In [2]:
 import pyforest
pyforest.lazy_imports()

['import lightgbm as lgb',
 'from scipy import stats',
 'from fbprophet import Prophet',
 'import gensim',
 'from statsmodels.tsa.arima_model import ARIMA',
 'from sklearn.preprocessing import OneHotEncoder',
 'from sklearn import metrics',
 'import re',
 'import pickle',
 'from PIL import Image',
 'import nltk',
 'import cv2',
 'import fbprophet',
 'from sklearn.ensemble import GradientBoostingRegressor',
 'import plotly.graph_objs as go',
 'import tqdm',
 'import imutils',
 'import keras',
 'import torch',
 'from pathlib import Path',
 'import sys',
 'from sklearn.linear_model import ElasticNetCV',
 'from sklearn.model_selection import StratifiedKFold',
 'import xgboost as xgb',
 'import textblob',
 'from sklearn.model_selection import KFold',
 'from dask import dataframe as dd',
 'from sklearn.model_selection import train_test_split',
 'from sklearn.preprocessing import LabelEncoder',
 'from sklearn.preprocessing import StandardScaler',
 'from sklearn.impute import SimpleImputer',
 

# Case 1 Estimating the population mean when sigma is known

Estimate the average age of all Mtech students admitted since 2020.The random sample data of 100 students collected and got
the values as sample mean as 30yrs.Historical evidence shows that the standard deviation of age of the MTech students are 6 yrs.Consider error rate as 5% or confidence level as 95%

In [7]:
n=100 # sample size
xbar=30 #sample means
sigma=6 #population std deviation
alpha=0.05 #confidence level=95%
std_error=sigma/np.sqrt(n)
zalphaby2=stats.norm.isf(alpha/2)
margin_error=zalphaby2*std_error
lower_CI=xbar-margin_error
upper_CI=xbar+margin_error
print(lower_CI,upper_CI)

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

28.824021609275967 31.175978390724033


In [39]:
stats.norm.interval(1-alpha,xbar,sigma/np.sqrt(n))

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

(28.45450241787066, 31.54549758212934)

Interpretation:
Range of Mean Age:

We are 95% confident that the true average age of all M.Tech students admitted since 2020 lies between 28.82 years and 31.18 years.
Inference:

The confidence interval reflects the precision of the estimate. A narrow interval indicates a high level of precision due to the large sample size (
𝑛
=100 n=100).

# Case 2 Estimating the population mean when sigma is uknown¶

Estimate the average age of all Mtech students admitted since 2020.The random sample data of 100 students collected and got the values as sample mean as 30yrs and standard deviation is 10yrs.Consider error rate as 1% or confidence level as 99%



In [15]:
n=100 # sample size
dof=n-1 #degrees of freedom
xbar=30 #sample means
alpha=0.01 #confidence level=99%
s=10
std_error=s/np.sqrt(n)
talphaby2=stats.t.isf(alpha/2,dof)
margin_error=talphaby2*std_error
lower_CI=xbar-margin_error
upper_CI=xbar+margin_error
print(lower_CI,upper_CI)

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

27.37359454271917 32.626405457280825


In [17]:
stats.t.interval(1-alpha,dof,xbar,sigma/np.sqrt(n))

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

(28.424156725631505, 31.575843274368495)

The output of the calculation, the confidence interval, infers the range within which we are 99% confident that the true mean age (
𝜇
μ) of all M.Tech students admitted since 2020 lies. Here's what it means in simple terms:

Precision of Estimate:
The confidence interval provides a margin of error around the sample mean (
𝑥
ˉ
x
ˉ
 ) to account for variability in the data. The width of the interval reflects how precise the estimate is.

Confidence Level:
A 99% confidence level means that if we were to repeat this sampling process many times, 99 out of 100 such confidence intervals would contain the true mean age of all M.Tech students.

Implication of the Output:

If the interval is narrow, the estimate of the mean age is precise.
If the interval is wide, there is more uncertainty about the true mean age.
The mean age of all M.Tech students is likely within this interval, considering the given sample.
For example, if the confidence interval turns out to be 
[
27.42
,
32.58
]
, we can infer that the true average age of all M.Tech students admitted since 2020 is likely between 

27.42 and 

32.58 years, with 99% confidence.


# Case 3:Estimate population proportion(P)

In random sample survey of traffic cases in Bangalore it is found that out of 100 cases,30 cases are drink and drive.At 99% confidence  level can you estimate what is the % of people drink and drive in bangalore?

In [28]:
alpha=0.01 #confidence level 99%
x=30# drink and drive cases
n=100 # sample size
p=x/n #sample proportion
std_error=np.sqrt(p*(1-p)/n)
zalphaby2=stats.norm.isf(alpha/2)
margin_error=zalphaby2*std_error
lower_CI=p-margin_error
upper_CI=p+margin_error
print(lower_CI,upper_CI)

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

0.18196067239201777 0.41803932760798224


Interpretation:
Range of Proportion:

We are 99% confident that the true proportion of drink-and-drive cases in Bangalore lies between 18.2% and 41.8%.
Inference:

This interval reflects the uncertainty due to sampling. The true percentage of drink-and-drive cases in Bangalore is likely within this range.
A wider confidence interval (like this one) indicates that there is substantial variability in the sample, possibly because the sample size is relatively small.

In [31]:
stats.norm.interval(1-alpha,p,np.sqrt(p*(1-p)/n))

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

(0.1819606723920178, 0.4180393276079822)

In [35]:
from statsmodels.stats.proportion import proportion_confint
proportion_confint(x,n,alpha)

(0.18196067239201777, 0.41803932760798224)

# Hypothesis Testing

In [44]:
df=sns.load_dataset('tips')

<IPython.core.display.Javascript object>

In [46]:
df.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

Business problem-1:
The restaurant manager claims that the average total bill per transaction is 20$.Validate the claim at 95% level of significance.[we assume 95% level confidence and try to see if that is true or not]

In [50]:
df.total_bill.mean()

19.78594262295082

# Step 1:Define H0 & H1 #Direction less or 2 tailed test
# H0: Mu_totalbill=20
# H1: Mu_totalbill !=0

# #Step2:Define confidence level (1-alpha)


In [75]:
alpha=0.05

# # Step 3: select the appropriate test and find the corresponding test statistics
# since the variable is single or one numerical variable --> One sample tets of mean.
# since unknown sigma (SD) one sample t test --> t-Statistic


In [78]:
xbar=df.total_bill.mean()# sample mean
s=df.total_bill.std() #sample std dev
mu=20 #assumed population mean
n=len(df.total_bill)
std_error=s/np.sqrt(n)
t_stat=(xbar-mu)/std_error
t_stat

<IPython.core.display.Javascript object>

-0.37559294451919506

# Step 4:Find the critical values :1 talphaby2, 2 alpha, 3 confidence level

In [80]:
dof=n-1
talphaby2=stats.t.ppf(alpha/2,dof)
talphaby2

<IPython.core.display.Javascript object>

-1.9697743954258797

In [82]:
# Step 5 Decision making  Critical value approach(CVA)
# Decision rule:if abs(t_stat)>abs(tcritical) ---> Reject H0 else Accept H0;

In [84]:
## since abs(-0.37559)=0.375 <abs(tcritical)=1.96 ===>accept H0;

In [86]:
# if H0 is accepted that means that we failed to reject the null hypothesis,from the given sample selected from population

# Confidence Interval Approach(CIA)

In [92]:
stats.t.interval(1-alpha,dof,xbar,s/np.sqrt(n))

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

(18.663331704358473, 20.908553541543167)

# P value Approach --Probability of null hypothesis being true
# Decision Rule:If the hypothesis value lies b/w confidence interval==> accept H0 else Reject H0

In [96]:
p_value=stats.t.sf(0.37559294451919506,dof)*2
p_value

<IPython.core.display.Javascript object>

0.7075471935626723

In [98]:
## Decision rule:if the p value >alpha -->Accept H0,else Reject H0;

In [100]:
stats.ttest_1samp(df.total_bill,mu)

<IPython.core.display.Javascript object>

TtestResult(statistic=-0.37559294451919506, pvalue=0.7075471935626723, df=243)